crates/nu-path/src/dots.rs

```diff
@@ -228,7 +294,7 @@ mod tests {
 
         #[test]
         fn string_with_three_ndots_and_garbage() {
-            check_ndots_expansion(r"ls ..\../ garbage.*[", "ls .../ garbage.*[");
+            check_ndots_expansion(r"file .../ garbage.*[", "file .../ garbage.*[");
```

Comment:
Anything we pass to an external command needs to already be expanded, as that's part of what the commands will expect the shell to do for them.

Reply:
Specifically, I think in this case `...` isn't part of any posix standard (that I'm aware of) so that functionality would need some special care when being handed to external commands.

Reply:
The `"ls .../ garbage.*["` is passed into `expand_ndots` as a single string. `expand_ndots` isn't able to determine if something is a command or a filepath. When something ends up in this function that should be treated separately as a command and a filepath,  they shouldn't be passed as a single string, they should be passed separately and other parts of nushell should take care of that. Because of this, I renamed the testcase, to not give the reader wrong expectations of the behavior of `expand_ndots`. From my testing, the `expand_ndots` function now behaves correctly, and I haven't seen nushell rely on the behavior of `expand_ndots` trying to treat commands differently.

Reply:
Thinking about this a bit more. Other variations of the `"ls .../ garbage.*["` example were handled differently in the previous implementation. `"ls ... garbage.*["`, this wouldn't have been expanded in the previous implementation, because there was no `/` after the `...`.  Point being that `expand_ndots` can't and shouldn't need to figure out if something is just a path or a command followed by a path. In my opinion, `expand_ndots` should treat the string that is given to it as a path and expand it accordingly. Behavior that was deviating from this in the previous implementation was accidental and also inconsistent. The change in this PR simply changes the oversight that `...` can appear at the end of a filename, in which case it shouldn't be expanded.

Reply:
@jntrnr Sorry for the inconvenience, but I have written down my opinion on the `expand_ndots` topic in the 2 posts above. Now someone that has the bigger picture of nushell in mind should make a decision if this PR should be merged, changed or closed.